test/src/test/java/org/corfudb/integration/StreamIT.java

```diff
@@ -42,8 +74,8 @@ public void simpleStreamTest() throws Exception {
         IStreamView s1 = rt.getStreamsView().get(streamId);
 
         // Verify that the stream is empty
-        assertThat(s1.hasNext())
-                .isFalse();
+        final int N_HASNEXT_TESTS = 20;
+        IntStream.range(0, N_HASNEXT_TESTS).forEach(n -> assertThat(s1.hasNext()).isFalse());
```

Comment:
hasNext seems to be unreliable and needs fixing. We can assert the results based on `streamUpTo()` call.